### ISSUE TITLE
Update thirdparty/ directory ABOUT files to latest SPEC

### ISSUE THREAD

**Issue by tdruez** [aboutcode-org/aboutcode-toolkit#296]
```
about-code check --show-all ./thirdparty
Running aboutcode-toolkit version 3.0.0
Checking ABOUT files...
INFO: base/appdirs.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/appdirs.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/certifi.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/certifi.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/chardet.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/chardet.ABOUT: Field license_text_url is not a supported field and is ignored.
INFO: base/chardet.ABOUT: Field license_text_file is not a supported field and is ignored.
CRITICAL: base/idna.ABOUT: Cannot load invalid ABOUT file: '/Volumes/nexB/aboutcode-toolkit/thirdparty/base/idna.ABOUT': AttributeError("'list' object has no attribute 'strip'",)
'list' object has no attribute 'strip'
INFO: base/packaging.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/packaging.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/packaging.ABOUT: Field notice_text_file is not a supported field and is ignored.
INFO: base/pip.ABOUT: Field author_file is not a supported field and is ignored.
INFO: base/pip.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/pip.ABOUT: Field license_text_file is not a supported field and is ignored.
CRITICAL: base/pyparsing.ABOUT: Cannot load invalid ABOUT file: '/Volumes/nexB/aboutcode-toolkit/thirdparty/base/pyparsing.ABOUT': ScannerError()
mapping values are not allowed here
  in "<unicode string>", line 5, column 27:
    download_url: download_url: https://pypi.python.org/packag ...
                              ^
INFO: base/requests.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/requests.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/setuptools.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/setuptools.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/six.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/six.ABOUT: Field license_text_file is not a supported field and is ignored.
CRITICAL: base/urllib3.ABOUT: Cannot load invalid ABOUT file: '/Volumes/nexB/aboutcode-toolkit/thirdparty/base/urllib3.ABOUT': ScannerError()
while scanning a quoted scalar
  in "<unicode string>", line 3, column 15:
    download_url: "https://pypi.python.org/package ...
                  ^
found unexpected end of stream
  in "<unicode string>", line 15, column 1:

    ^
INFO: base/virtualenv.py.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/virtualenv.py.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/wheel.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/wheel.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: base/wincertstore.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: base/wincertstore.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: dev/apipkg.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: dev/apipkg.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: dev/funcsigs.ABOUT: Field homepage is not a supported field and is ignored.
INFO: dev/mock.ABOUT: Field homepage is not a supported field and is ignored.
INFO: dev/pbr.ABOUT: Field homepage is not a supported field and is ignored.
INFO: dev/py.ABOUT: Field contatct is not a supported field and is ignored.
INFO: dev/py.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: dev/py.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: dev/py.ABOUT: Field vcs_repo is not a supported field and is ignored.
CRITICAL: dev/pytest.ABOUT: Cannot load invalid ABOUT file: '/Volumes/nexB/aboutcode-toolkit/thirdparty/dev/pytest.ABOUT': ScannerError()
mapping values are not allowed here
  in "<unicode string>", line 6, column 20:
    description: pytest: simple powerful testing with P ...
                       ^
INFO: prod/backports.csv.ABOUT: Field homepage is not a supported field and is ignored.
INFO: prod/boolean.py.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/boolean.py.ABOUT: Field license_text_file is not a supported field and is ignored.
CRITICAL: prod/click.ABOUT: Cannot load invalid ABOUT file: '/Volumes/nexB/aboutcode-toolkit/thirdparty/prod/click.ABOUT': ScannerError()
while scanning a simple key
  in "<unicode string>", line 8, column 1:
    homepage:http://click.pocoo.org/
    ^
could not find expected ':'
  in "<unicode string>", line 9, column 1:
    vcs_tool: git
    ^
INFO: prod/colorama.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/colorama.ABOUT: Field keywords is not a supported field and is ignored.
INFO: prod/colorama.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: prod/Jinja2.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/Jinja2.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: prod/license_expression.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/license_expression.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: prod/MarkupSafe.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/MarkupSafe.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: prod/pyyaml.py2.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/pyyaml.py2.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: prod/pyyaml.py3.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/pyyaml.py3.ABOUT: Field license_text_file is not a supported field and is ignored.
INFO: prod/pyyaml.tar.ABOUT: Field dje_license is not a supported field and is ignored.
INFO: prod/pyyaml.tar.ABOUT: Field license_text_file is not a supported field and is ignored.
Found 57 errors.
```

**Comment by tdruez** [comment]
Once this is fixed, it would make sense to add the following to the CI config:
```
before_script:
    about-code check --show-all ./thirdparty/
```

**Comment by tdruez** [comment]
@chinyeungli there's still a couple issues on my side:
```
python3.6 -m 'attributecode.cmd' check --show-all ./thirdparty
Running aboutcode-toolkit version 3.0.0
Checking ABOUT files...
INFO: base/pip.ABOUT: Field author_file is not a supported field and is ignored.
INFO: prod/colorama.ABOUT: Field keywords is not a supported field and is ignored.
Found 2 errors.
```

Also, what's your take on https://github.com/nexB/aboutcode-toolkit/issues/296#issuecomment-344483085 ?

**Comment by chinyeungli** [comment]
Well, the thing is the `author_file` and the `keywords` is not not the supported field.
Do you think we should add the `--mapping` option so that we can include these fields as a custom field?

**Comment by tdruez** [comment]
I think non-supported fields should not be part of the ABOUT files. No need for `--mapping`.

**Comment by chinyeungli** [comment]
You mean to remove the `author_file` field and the `keyword` field in the 2 .ABOUT files log in the error?

**Comment by tdruez** [comment]
Yes, I don't see the point to maintain values for non-supported fields.

**Comment by chinyeungli** [comment]
Well, for some users, they may want to have some information of their own
in About files such as "audit reference number", "internal notes" etc.. but
I agree we should not keep the non-supported fields for our own About
files.

On Dec 8, 2017 11:19 AM, "tdruez" <notifications@github.com> wrote:

Yes, I don't see the point to maintain values for non-supported fields.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/nexB/aboutcode-toolkit/issues/296#issuecomment-350161441>,
or mute the thread

**Comment by tdruez** [comment]
Right, my comment applies to the ABOUT files in the `aboutcode-toolkit` codebase.

**Comment by tdruez** [comment]
@chinyeungli everything looks good now:
```
aboutcode-toolkit/ $ about-code check --show-all ./thirdparty
Running aboutcode-toolkit version 3.0.2
Checking ABOUT files...
No error found.
```

I thing we should also add that check as part of the continuous integration, like we did in https://github.com/nexB/license-expression/commit/0d0ac506b64bb00e91d7cd504d6656d627121e2a

It's also a good time to add a section in the documentation about adding `about-code` in your CI.

**Comment by pombredanne** [comment]
This has been fixed: I updated the thirdparty with #355 and ensure all ABOUT files were correct.

**Comment by pombredanne** [comment]
Also in a branch  with this commit, I am adding built-ins tests for ABOUT files https://github.com/nexB/aboutcode-toolkit/blob/60bd5e7adce79ff46868b460095ecbd39f52b1ae/tests/test_validate.py
